packages/core/src/analytics/redux/middlewares/bag.js

```diff
@@ -278,34 +277,31 @@ export default (analyticsInstance, customActionTypes) => {
           oldQuantity: previousBagItem?.quantity,
           ...getBagData(action),
         };
-        let eventType = null;
-
-        // Track ga4 update event
-        // Data is being cloned here because it will be mutated later
-        analyticsInstance.track(eventTypes.PRODUCT_UPDATED, { ...data });
-
-        if (data.oldSize && data.oldSize !== data.size) {
-          const removedProductData = { ...data };
-          removedProductData.size = data.oldSize;
-          removedProductData.quantity = data.oldQuantity;
-
-          analyticsInstance.track(
-            eventTypes.PRODUCT_REMOVED_FROM_CART,
-            removedProductData,
-          );
 
-          data.oldQuantity = 0;
+        // product updated will only triggered if provenience is from BAG
+        if (data.from === fromParameterTypes.BAG) {
+          // Track ga4 update event
```

Comment:
(this can be removed - product updated if for any integration that has interest on the event)
```suggestion
```